src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -175,11 +185,34 @@ public Task StopAsync()
             catch (Exception ex)
             {
                 Log.UnexpectedExceptionInStop(Logger, ConnectionId, ex);
-            }
-            
+            }
+
             return Task.CompletedTask;
         }
 
+        public async Task CloseAsync()
+        {
+            if (!_closeConnectionLock.Wait(0))

```

Comment:
If it's only ever called once, you don't need the lock

Reply:
Already replaced by 'Interlock', I just want to prevent it from being called twice or more.

Reply:
So it is either called once and then we don't need any kind of lock. Or it could be called multiple times and then do the callers always wait for the returned task? If they do I'd like to see us returning the same task for each caller.

Reply:
Since `Interlock.Exchange` is lock-free, I guess it's enough?